storage/azure/client.go

```diff
@@ -0,0 +1,73 @@
+package azure
+
+import (
+	"bytes"
+	"context"
+	"errors"
+	"fmt"
+	"os"
+
+	"github.com/Azure/azure-sdk-for-go/sdk/azcore"
+	"github.com/Azure/azure-sdk-for-go/sdk/storage/azblob"
+)
+
+type Client interface {
+	// Read an object from an Azure blob.
+	Read(ctx context.Context, container, blob string) ([]byte, error)
+
+	// Write an object onto an Azure blob.
+	Write(ctx context.Context, container, blob string, p []byte) error
+}
+
+type client struct {
+	BlobAPI *azblob.Client
+}
+
+// newClient returns a new instance of Client.
+func newClient(config *Config) (Client, error) {
+	// If the access key isn't defined in the configuration, try to read it from the environment.
+	if config.AccessKey == "" {
+		config.AccessKey = os.Getenv("ARM_ACCESS_KEY")
+	}
+
+	cred, err := azblob.NewSharedKeyCredential(config.AccountName, config.AccessKey)
+	if err != nil {
+		return nil, err
+	}
+
+	url := fmt.Sprintf("https://%s.blob.core.windows.net/", config.AccountName)
+	c, err := azblob.NewClientWithSharedKeyCredential(url, cred, nil)
+
+	return &client{c}, err
+}
+
+// Read an object from an Azure blob.
+func (c *client) Read(ctx context.Context, container, blob string) ([]byte, error) {
+	resp, err := c.BlobAPI.DownloadStream(ctx, container, blob, nil)
+	if err != nil {
+		var responseError *azcore.ResponseError
+		if errors.As(err, &responseError) {
+			if responseError.StatusCode == 404 {
+				val := []byte{}
+				return val, c.Write(ctx, container, blob, val)
```

Comment:
Sorry that my words were unclear, but if the history file doesn't exist, don't write an empty file to remote storage; instead, return empty bytes to the caller without error.

Reply:
My apologies for the misunderstanding—fixed in https://github.com/minamijoyo/tfmigrate/pull/173/commits/7d3c274278f684433fa500360e34e2af05a870e4.